local/bin/py/build/pull_rbac.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+import sys
+import requests
+import json
+from os import getenv
+
+def pull_rbac():
+  api_endpoint = 'https://app.datadoghq.com/api/v2/permissions'
+  headers = {'DD-API-KEY': sys.argv[1], 'DD-APPLICATION-KEY': sys.argv[2]}
+  formatted_permissions_dict = {}
+
+  r = requests.get(api_endpoint, headers=headers)
+
+  if r.status_code == requests.codes.ok:
+    json_result = r.json()
+    data = json_result.get('data', []);
+
+    for permission in data:
+      group_name = permission['attributes']['group_name']
+      permission_name = permission['attributes']['name']
+
+      # Remove legacy logs permissions from dictionary before converting to JSON.  These legacy permissions are hard-coded in rbac-permissions-table partial until they can be deprecated.
+      if permission_name in ('logs_live_tail', 'logs_read_index_data'):
+        del permission
+      else:
+        if group_name not in formatted_permissions_dict.keys():
+          formatted_permissions_dict[group_name] = [permission]
```

Comment:
you can use a default dict here to ensure all elements are initialized with a empty array

```
>>> s = [('yellow', 1), ('blue', 2), ('yellow', 3), ('blue', 4), ('red', 1)]
>>> d = defaultdict(list)
>>> d.default_factory
<type 'list'>
>>> for k, v in s:
...     d[k].append(v)
>>> d.items()
[('blue', [2, 4]), ('red', [1]), ('yellow', [1, 3])]
>>> d
defaultdict(<type 'list'>, {'blue': [2, 4], 'red': [1], 'yellow': [1, 3]})
```

Reply:
hey @pietdaniel thanks so much for the review.   this is my first time writing python so i may have misunderstood but i didn't see the advantage of declaring a `defaultdict` here without having keys available first.  looking at the docs for `defaultdict` i was able to use `setdefault` though which seems cleaner than what I had before.   let me know if you have any concerns here.   thanks!